### DISCUSSION TITLE
Construction Time-lapse with GoPro Black Hero 8

### DISCUSSION THREAD

**Discussion by madgiordano** [gopro/labs#267] [Q&A]
I'm setting up a GoPro 8 hero camera underwater and need to take pictures of changing conditions over time. My original set up uses external batteries and a time-lapse camera controller (Blink-X). I would like to take three photos a day, preferably around midday. The reason for so few pictures is to maintain battery life for extended deployment. I learned recently that I may be able to eliminate the time-lapse controller by overriding the GoPro software using the construction time-lapse schedule but have been unsuccessful. Ideally, camera will be collected after month-ish deployments. The camera has either turned on independently (at the correct time) and taken continuous photos at short intervals until it dies or seems to be taking pictures but when I go into photos, nothing is actually there. Any advice? Thanks!

**Comment by dnewman-gpsw** [comment]
I have done 45 day, 3-photo per day timelapse, on the internal battery. This is straight forward.  So three photo arround at midday?  Something like:

![image](https://user-images.githubusercontent.com/10856084/225724856-69fe2840-8343-494c-96df-a9379ec2b9d0.png)

This command steps.  
mP - mode Photo
dP - default Photo 
!11:00S - sleep into 11:00, wake and press Shutter
!12:00S - sleep into 12:00, wake and press Shutter
!13:00S - sleep into 13:00, wake and press Shutter
!R - Repeat

This is only using the basic scripting commands that should work on all GoPro models.

**Reply by madgiordano** [reply]
Thanks so much! It works great on my Hero Black 8. Last quick question, is there any way to get this script to work on a Hero 5?

**Reply by dnewman-gpsw** [reply]
Sorry, Labs was never ported to HERO5, HERO7 onward have Labs in any real form.

**Reply by madgiordano** [reply]
Got it. Thanks!

**Reply by madgiordano** [reply]
Follow up question. Programming seems not to work when plugged in. The camera will last on its own with this program for about 2 weeks. I was trying to hook it up to an external battery for long-term deployment, but I'm not sure if that's possible. Any ideas?

**Reply by dnewman-gpsw** [reply]
programming?  Do you mean reading QR Code, that should not care whether the camera is plugged in.  External power is possible and common, just remove to internal battery to run the above code.

**Reply by madgiordano** [reply]
Is there a newer camera that has this type of capability built in that you might recommend? Or is the best solution the Labs software? I haven't been able to get it to work even without the battery inserted.

**Reply by madgiordano** [reply]
I recently bought a Hero Black 12. When trying to run this same program on the new camera I've found that the camera is not turning off after taking the pictures, but it is taking them at the right time, it will just die within a day since it is not turning off. Are you aware of any changes I could make in the QR code to ensure the camera understands to turn off between takes?
Thanks so much for all your help!

**Reply by dnewman-gpsw** [reply]
That makes no sense, that command should work perfectly all Labs cameras. I just tried on my HERO12, it shut off correctly in response.  You should be able to see it is working within seconds of scanning that code.  Share you log file from MISC/qrlog.txt

**Reply by madgiordano** [reply]
Thanks for trying it on your 12! I wonder what my issue could be. I reset my camera last night so I could show you the log without any complications from my attempted troubleshooting. I read the QR at 16:33 2/20/2024 with 85% battery and did not touch it until after a full cycle today, after the camera had died. Here is the log:
[qrlog.txt](https://github.com/gopro/labs/files/14365250/qrlog.txt)

Another oddity I've noticed when trying to run this program on the Black 12 is that it won't attempt to start in the middle of a cycle like the other cameras I've run on the program. It will always notify me it will start the next day.

Here is what a first day run looks like on a camera that looks like it's running properly:
[qrlog-Black7 (working).txt](https://github.com/gopro/labs/files/14365271/qrlog-Black7.working.txt)

**Reply by dnewman-gpsw** [reply]
Just observe what your camera is doing.  Run experiments around the time is it. So if is 4pm, try mPdP!16:02S!16:04S!16:06S

As it works on other HERO12, you need to see if you might need to factory reset this unit.

If you see it isn't turning off, you might learn why.  It is probably asking a simple popup question (which can be solved.) New cameras ask a lot of user onboarding questions, can be annoying. 

Make sure is it switching to photo, and that you get photos on the SD card.

Re:  it won't attempt to start in the middle of a cycle like the other cameras I've run on the program.

It is running the program the same only all models. This script is literally says sleep until 11am as its first command, if it was 11:05, it would sleep 23h 55m on all cameras.
